File: Singleton/Java/src/dev/ismail/SingletonExample.java
package dev.ismail;

public class SingletonExample {

    private static SingletonExample singleTonInstance = null;


    private SingletonExample() {

    }

    public static SingletonExample getInstance() {
        if(singleTonInstance == null) singleTonInstance = new SingletonExample();
        return singleTonInstance;
    }

}

File: Singleton/PHP/SingletonExample.php
<?php


class SingletonExample
{

    private static $instance = null;

    private function __construct()
    {

    }

    public static function  getInstance(){
        if (!isset($instance)){
            $instance = new SingletonExample();
        }
        return $instance;
    }


    public function __toString()
    {
       return spl_object_hash($this)  ;
    }
}